src/metabase/search/models/search_index_metadata.clj

```diff
@@ -59,17 +64,17 @@
 (defn delete-index!
   "Delete the given pending index, as long as its still pending."
   [engine version index-name]
-  (t2/delete! :model/SearchIndexMetadata :engine engine :version version :index_name (name index-name)))
+  (t2/delete! :model/SearchIndexMetadata :engine engine :version version :lang_code (i18n/site-locale-string) :index_name (name index-name)))
 
 (defn active-pending!
   "If there is 'pending' index, make it 'active'. Return the name of the active index, regardless."
   [engine version]
   (t2/with-transaction [_conn]
-    (when (t2/exists? :model/SearchIndexMetadata :engine engine :version version :status :pending)
-      (t2/delete! :model/SearchIndexMetadata :engine engine :version version :status :retired)
-      (t2/update! :model/SearchIndexMetadata {:engine engine :version version :status :active} {:status :retired})
-      (t2/update! :model/SearchIndexMetadata {:engine engine :version version :status :pending} {:status :active}))
-    (t2/select-one-fn :index_name :model/SearchIndexMetadata :engine engine :version version :status :active)))
+    (when (t2/exists? :model/SearchIndexMetadata :engine engine :version version :lang_code (i18n/site-locale-string) :status :pending)
+      (t2/delete! :model/SearchIndexMetadata :engine engine :version version :lang_code (i18n/site-locale-string) :status :retired)
+      (t2/update! :model/SearchIndexMetadata {:engine engine :version version :lang_code (i18n/site-locale-string) :status :active} {:status :retired})
+      (t2/update! :model/SearchIndexMetadata {:engine engine :version version :lang_code (i18n/site-locale-string) :status :pending} {:status :active}))
+    (t2/select-one-fn :index_name :model/SearchIndexMetadata :engine engine :version version :lang_code (i18n/site-locale-string) :status :active)))
```

Comment:
I'd say it's worth binding lang code to a local variable as a "best practice":

1. Safeguard against (infinitesimally likely) concurrent changes to the value.
3. Save the (minor) overhead of repeated calls.